features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListPresenter.kt

```diff
@@ -71,20 +77,37 @@ class RoomListPresenter @Inject constructor(
             initialLoad(matrixUser)
         }
 
+        val sessionVerificationIsReady by sessionVerificationService.isReady.collectAsState()
+        val verificationState by sessionVerificationService.verificationAttemptStatus.collectAsState()
+        val presentVerificationSuccessfulMessage = remember(verificationState) {
+            derivedStateOf { verificationState == SessionVerificationServiceState.Finished }
+        }
+
+        LaunchedEffect(sessionVerificationIsReady) {
```

Comment:
Why are you using a LaunchedEffect here?

Reply:
We need it to safely access `.value` in the StateFlow instead of using `collectAsState()`, but it's a bit messy. Is there a better way to only collect that state if `sessionVerificationIsReady` is true? It should be safe to use the default value, but technically, the `isVerified` flow is only valid after the service is ready.

Reply:
If you do like this, the `isVerified` won't be reactive then

Reply:
I'd let the isVerified flow be smart enough with some flatMap inside the service?

Reply:
I simplified it a bit, also changed the true/false state to `Unknown, `NotVerified`, `Verified`.